app/models/condition.rb

```diff
@@ -0,0 +1,7 @@
+class Condition < ApplicationRecord
+  has_paper_trail
+
+  belongs_to :routing_page, class_name: "Page", foreign_key: "routing_page_id"
+  belongs_to :check_page, class_name: "Page", optional: true, foreign_key: "check_page_id"
+  belongs_to :goto_page, class_name: "Page", optional: true, foreign_key: "goto_page_id"
```

Comment:
The idea here is that for a Condition object, you can call `c.routing_page` to find out which page it's attached to, `c.check_page` to find the page that it's checking, and `c.goto_page` to find the page that it skips to